synapse/handlers/auth.py

```diff
@@ -617,6 +617,17 @@ def add_threepid(self, user_id, medium, address, validated_at):
             self.hs.get_clock().time_msec()
         )
 
+    @defer.inlineCallbacks
+    def delete_threepid(self, user_id, medium, address):
+        # 'Canonicalise' email addresses as per above
```

Comment:
In this case, the msisdn canonicalisation will have been done by the ID server (or the thing doing the verification). In fact, with us now having decided to have a separate thing to do the actual validation of 3pids, the email canonicalisation would go there too (it's a bit tricky for email given that squelching the localpart down to lowercase is not technically valid).